### PR TITLE
[github-actions] move POSIX expects to POSIX workflow

### PR DESCRIPTION
**Pull request by bukepo**

### CODE REVIEW

File: .github/workflows/posix.yml [hunk 1]
```diff
@@ -243,6 +315,8 @@ jobs:
 
   upload-coverage:
     needs:
+    - expects-macOS
```
Comment (Moandor-y) [inline .github/workflows/posix.yml:318]: We need to add uploading steps (like [line 93](https://github.com/openthread/openthread/pull/6195/files#diff-258c8e6cdf61e87e9d94e06c76fb5e23f602423313565dcb8a46afe648587aceR93)) in `expects-macOS` if we are going to upload its coverage data.

Comment (bukepo) [reply .github/workflows/posix.yml:318]: I failed to enable coverage on macOS, and we can try it later.

Comment (Moandor-y) [reply .github/workflows/posix.yml:318]: Then I think we should remove this dependency first, and add it when coverage on macOS is ready.

**Review by Moandor-y — APPROVED** [review]
LGTM :+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**